src/main/java/org/dswarm/xsd2jsonschema/JsonSchemaParser.java

```diff
@@ -76,6 +76,7 @@
 	//private static final XSImplementation impl         = (XSImplementation) new XSImplementationImpl();
 
 	private static final XSLoader LOADER;
+	private static final int MAX_RECURSION_DEPTH = 7;
```

Comment:
is it possible to provide this attribute/information via a parameter? - since, "7" seems to be rather arbitrary ;)